### ISSUE TITLE
Alacritty does not report BELs to X11

### ISSUE THREAD

**Issue by Celti** [alacritty/alacritty#1072]
I realise that this is perhaps a *desired* feature for many people, who just use Alacritty's built-in visual bell and are happy about it, but a visual bell built into the terminal is, well, only noticed when you're looking at the terminal.

For quite a while now, I've been using [xvisbell](https://github.com/rianhunter/xvisbell) to flash the *entire* screen (not just my terminal) when an X11 bell is received, and further supplementing that using xkbevd to play an audio signal if a bell is received and the system is idle or the screen is locked (i.e., I still am signalled even when not paying attention to the system and won't see a visual bell).

Alacritty unfortunately breaks that workflow. Would a PR that modifies the `bell()` method on Handler to report them to X11 (in addition to the built-in visual bell functionality) be accepted?

**Comment by jwilm** [comment]
What is necessary to report them to X11? We already raise X's "urgent" flag on the terminal window when a bell is received.

**Comment by Celti** [comment]
Sending an XKB bell event. Essentially, this would just call `xcb::xkb::bell()` (or whatever appropriate X11 library crate) on X11. Using the urgency hint is orthogonal to that, and I can rework my environment based on that, but it's generally expected that a terminal emulator on X11 will send that event.

**Comment by chrisduerr** [comment]
Since we now support running commands on bell, users should be able to do this themselves.

**Comment by tinywrkb** [comment]
@chrisduerr I hope that I'm not wrong about this but when I tested the bell command it looks like it's not a child process of the Alacritty window but it's detached, so is a child process of the init / PID 1.  
The compositor needs to know the PID of the Alacritty window in order to set the urgent hint so it's not possible to use something like this

```
bell:
    program: swaymsg
    args: ["[pid=$(ps -o pid=|head -1)] urgent enable"]
```

I haven't reported this because I wanted to test it again, though I did confirm it with pstree.  
Can anyone else concur?
